presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -974,28 +974,35 @@ public void testEnum()
         }
     }
 
+    /**
+     * @see #testTimeCoercion
+     */
     @Test(dataProvider = "testTimestampDataProvider")
     public void testTime(boolean insertWithPresto, ZoneId sessionZone)
     {
         LocalTime timeGapInJvmZone = LocalTime.of(0, 12, 34, 567_000_000);
         checkIsGap(jvmZone, timeGapInJvmZone.atDate(EPOCH_DAY));
 
         DataTypeTest tests = DataTypeTest.create()
-                .addRoundTrip(timeDataType(), LocalTime.of(1, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(2, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(2, 12, 34, 1_000_000))
-                .addRoundTrip(timeDataType(), LocalTime.of(3, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(4, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(5, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(6, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(9, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(10, 12, 34, 0))
-                .addRoundTrip(timeDataType(), LocalTime.of(15, 12, 34, 567_000_000))
-                .addRoundTrip(timeDataType(), LocalTime.of(23, 59, 59, 999_000_000));
-
-        // epoch is also a gap in JVM zone
-        tests.addRoundTrip(timeDataType(), epoch.toLocalTime());
-        tests.addRoundTrip(timeDataType(), timeGapInJvmZone);
+                .addRoundTrip(timeDataType(0), LocalTime.of(1, 12, 34, 0))
+                .addRoundTrip(timeDataType(1), LocalTime.of(2, 12, 34, 100_000_000))
+                .addRoundTrip(timeDataType(2), LocalTime.of(2, 12, 34, 10_000_000))
+                .addRoundTrip(timeDataType(3), LocalTime.of(2, 12, 34, 1_000_000))
+                .addRoundTrip(timeDataType(3), LocalTime.of(3, 12, 34, 0))
+                .addRoundTrip(timeDataType(4), LocalTime.of(4, 12, 34, 0))
+                .addRoundTrip(timeDataType(5), LocalTime.of(5, 12, 34, 0))
+                .addRoundTrip(timeDataType(5), LocalTime.of(6, 12, 34, 0))
+                .addRoundTrip(timeDataType(6), LocalTime.of(9, 12, 34, 0))
+                .addRoundTrip(timeDataType(6), LocalTime.of(10, 12, 34, 0))
+                .addRoundTrip(timeDataType(6), LocalTime.of(15, 12, 34, 567_000_000))
+                .addRoundTrip(timeDataType(6), LocalTime.of(23, 59, 59, 0))
+                .addRoundTrip(timeDataType(6), LocalTime.of(23, 59, 59, 999_000_000))
+                .addRoundTrip(timeDataType(6), LocalTime.of(23, 59, 59, 999_900_000))
+                .addRoundTrip(timeDataType(6), LocalTime.of(23, 59, 59, 999_990_000))
+                .addRoundTrip(timeDataType(6), LocalTime.of(23, 59, 59, 999_999_000))
+                // epoch is also a gap in JVM zone
```

Comment:
For later... we should fix this test to not use `testTimestampDataProvider`. Timezones don't play a role in TIME type, so we don't need to do any checks for gaps or different timezones.

Reply:
Yes, the do not play anymore, but i think testing with more than one session zone is still valuable from blackbox testing perspective.
(This is why testDate loops over time zones, even though date handling never dependent on a zone)
Just my point of view.

Reply:
That makes the tests misleading. Just because TIME used to be sensitive to timezones, it doesn't mean that we should continue to treat them that way, just like we don't think about or test other types such as decimal, bigint, etc, that way.

Reply:
You could same way argue that since we're using java.time (let's assume we do uniformly) we do not need to set non-UTC as test JVM zone -- the result does not depend on the zone. Yet, it was important in the past to test with non-UTC zone and thus it remains important even after the code is fixed, as a regression testing measures.
Similarly, testing date/time with session zones is reasonable from regression testing perspective.
Testing bigint or decimal cannot be argued the same way.

Reply:
If we had implemented TIMESTAMP and TIME right from the start, we would've never added these tests, just saying. Of course it makes sense for TIMESTAMP W/ TZ and TIME W/ TZ, but those are different types with different semantics.